skypy/galaxy/stellar_mass.py

```diff
@@ -0,0 +1,73 @@
+r"""Models of galaxy stellar mass.
+
+"""
+
+import numpy as np
+
+from skypy.utils.random import schechter
+
+
+__all__ = [
+    'schechter_mass',
+]
+
+
+def schechter_mass(alpha, m_star, x_min, x_max, resolution=100, size=None):
+    r""" Stellar masses following the Schechter mass function [1]_.
+
+    Parameters
+    ----------
+    alpha : float
+        The alpha parameter in the Schechter stellar mass function.
+    m_star : array-like
+        Characteristic stellar mass M_*.
+    size: int, optional
+         Output shape of stellar mass samples. If size is None and m_star
+         is a scalar, a single sample is returned. If size is None and
+         m_star is an array, an array of samples is returned with the same
+         shape as m_star.
+    x_min, x_max : float
+        Lower and upper bounds for the random variable x in units of M_*.
+    resolution : int, optional
+        Resolution of the inverse transform sampling spline. Default is 100.
+
+    Returns
+    -------
+    stellar mass : array_like
+        Drawn stellar masses from the Schechter stellar mass function in units
+        of the solar mass
+
+    Notes
+    -----
+     The stellar mass function is given by a Schechter function as
+
+    .. math::
+        \Phi(M, z) = \frac{\Phi_\star(z)}{M_\star}
+            \left(\frac{M}{M_\star}\right)^\alpha
+            \exp\left(-\frac{M}{M_\star}\right) \;.
```

Comment:
Not really relevant here since that is the number density, but the sampling does not draw galaxies from a Schechter function, it assigns masses to already drawn galaxies. This needs more explanation so that users understand what this function is doing and how it interacts with the actual sampling of galaxies, which currently happens in the redshift module.